src/argilla_sdk/datasets/_resource.py

```diff
@@ -166,6 +167,50 @@ def create(self) -> "Dataset":
     def from_model(cls, model: DatasetModel, client: "Argilla") -> "Dataset":
         return cls(client=client, _model=model)
 
+    def to_disk(self, path: str) -> str:
+        """Exports the dataset to disk in the given path.
+
+        Args:
+            path (str): The path to export the dataset to.
+        """
+
+        path = os.path.join(path, self.name)
+        os.makedirs(path, exist_ok=True)
+        dataset_path = os.path.join(path, "dataset.json")
+        settings_path = os.path.join(path, "settings.json")
+        records_path = os.path.join(path, "records.json")
+
+        with open(dataset_path, "w") as f:
+            json.dump(self._model.model_dump(), f)
+
+        self.settings.to_disk(settings_path)
+
+        if self.exists():
+            self.records.to_disk(records_path)
+
+        return path
+
+    @classmethod
+    def from_disk(
+        cls,
+        path: str,
+        client: Optional["Argilla"] = None,
+    ) -> "Dataset":
+        dataset_path = os.path.join(path, "dataset.json")
+
+        with open(dataset_path, "r") as f:
+            dataset_model = json.load(f)
+            dataset_model = DatasetModel(**dataset_model)
+        dataset = cls.from_model(model=dataset_model, client=client)
```

Comment:
What happens if the dataset does not exist?  I would expect this action to create the dataset. I would expect an error if the dataset already exists in Argilla.

Reply:
In the case of no dataset, I agree that it should create one
In the case of an existing dataset, we need to add a utility to deal with the name conflict. Either we raise an error that sames 'rename the existing ds' or we allow the user to name the incoming ds.